Detections/SigninLogs/NewCountryValidCreds.yaml

```diff
@@ -0,0 +1,62 @@
+id: 7808c05a-3afd-4d13-998a-a59e2297693f
+name: New country signIn with correct password
+description: |
+  'Identifies an interrupted sign-in session from a country the user has not sign-in before in the last 7 days, where the password was correct. Although the session is interrupted by other controls such as multi factor authentication or conditional access policies, the user credentials should be reset due to logs indicating a correct password was observed during sign-in.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs 
+queryFrequency: 1h
+queryPeriod: 7d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - CredentialAccess
+relevantTechniques:
+  - T1078
+  - T1110
+query: |
+  // Creating a list of successful sign-in by users in the last 7 days.
+  let KnownUserCountry = (
+  SigninLogs
+  | where TimeGenerated >= ago(7d)
+  | where ResultType == 0
+  | summarize KnownCountry = make_set(Location) by UserPrincipalName
+  );
+  // Identify sign-ins that are no successful but have the auth details indicating a correct password.
+  SigninLogs
+  | where TimeGenerated >= ago(1h)
+  | where ResultType != 0
+  | where parse_json(AuthenticationDetails)[0].authenticationMethod == "Password"
+  | extend PasswordResult = tostring(parse_json(AuthenticationDetails)[0].authenticationStepResultDetail)
+  | extend AuthSucceeded = tostring(parse_json(AuthenticationDetails)[0].succeeded)
```

Comment:
Parse_json on AuthenticationDetails can be used just once. This would improve the performance of the query.

Reply:
Changes have been applied with the new commit